packages/babel-plugin/src/utils/css-map.ts

```diff
@@ -0,0 +1,192 @@
+import type { EvaluateExpression } from './types';
+
+const createErrorMessage = (message?: string): string => {
+  return `
+${
+  message || 'Using a CSS Map in this manner is incorrect.'
+} To correctly implement a CSS Map, follow the syntax below:
+
+\`\`\`
+import { css, cssMap } from '@compiled/react';
+const borderStyleMap = cssMap({
+    none: { borderStyle: 'none' },
+    solid: { borderStyle: 'solid' },
+});
+const Component = ({ borderStyle }) => <div css={css(borderStyleMap[borderStyle])} />
+\`\`\`
+    `;
+};
+
+/**
+ * Retrieves the leftmost identity from a given expression.
+ *
+ * For example:
+ * Given a member expression "colors.primary.500", the function will return "colors".
+ *
+ * @param expression The expression to be evaluated.
+ * @returns {string} The leftmost identity in the expression.
+ */
+const findBindingIdentifier = (
+  expression: t.Expression | t.V8IntrinsicIdentifier
+): t.Identifier | undefined => {
+  if (t.isIdentifier(expression)) {
+    return expression;
+  } else if (t.isCallExpression(expression)) {
+    return findBindingIdentifier(expression.callee);
+  } else if (t.isMemberExpression(expression)) {
+    return findBindingIdentifier(expression.object);
+  }
+
+  return undefined;
+};
+
+/**
+ * Retrieves the CSS Map related information from a given expression.
+ *
+ * @param expression The expression to be evaluated.
+ * @param meta {Metadata} Useful metadata that can be used during the transformation
+ * @param evaluateExpression {EvaluateExpression} Function that evaluates an expression
+ */
+const getCSSmap = (
+  expression: t.Expression,
+  meta: Metadata,
+  evaluateExpression: EvaluateExpression
+):
+  | {
+      value: t.ObjectExpression;
+      meta: Metadata;
+      property: t.Identifier | t.StringLiteral;
+      computed: boolean;
+    }
+  | undefined => {
+  // Bail out early if cssMap callExpression doesn't exist in the file
+  if (!meta.state.compiledImports?.cssMap) return undefined;
+
+  // We only care about member expressions. e.g. variants[variant]
+  if (!t.isMemberExpression(expression)) return undefined;
+
+  const bindingIdentifier = findBindingIdentifier(expression.object);
+
+  if (!bindingIdentifier) return undefined;
+
+  // Evaluate the binding identifier to get the value of the CSS Map
+  const { value, meta: updatedMeta } = evaluateIdentifier(
+    bindingIdentifier,
+    meta,
+    evaluateExpression
+  );
+
+  // Ensure cssMap is used in a correct format.
+  if (
+    t.isCallExpression(value) &&
+    t.isIdentifier(value.callee) &&
+    value.callee.name === meta.state.compiledImports?.cssMap &&
+    value.arguments.length > 0 &&
+    t.isObjectExpression(value.arguments[0])
+  ) {
+    // It's CSS Map! We now need to check if the use of the CSS Map is correct.
+    if (t.isCallExpression(expression.object)) {
+      throw buildCodeFrameError(createErrorMessage(), expression, updatedMeta.parentPath);
```

Comment:
I think it would be better to be specific on these errors. Throwing the error on `expression.object` and saying why (call expression isn't supported) etc

Reply:
Good idea.